x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Values.java

```diff
@@ -84,7 +85,13 @@ public DataType dataType() {
 
     @Override
     protected TypeResolution resolveType() {
-        return EsqlTypeResolutions.isNotSpatial(field(), sourceText(), DEFAULT);
+        return TypeResolutions.isType(
+            field(),
+            dt -> DataType.isSpatial(dt) == false && dt != UNSIGNED_LONG,
+            sourceText(),
+            DEFAULT,
+            "any type except unsigned_long and spatial types"
+        );
```

Comment:
This was accepting unsigned_long, which failed later on toSupplier()

Reply:
:+1:
Yay tests catching things.